programs/uxd/tests/integration_tests/cases/test_mint_with_identity_depository.rs

```diff
@@ -0,0 +1,111 @@
+use solana_program::pubkey::Pubkey;
+use solana_program_test::ProgramTestContext;
+use solana_sdk::signature::Keypair;
+
+use crate::integration_tests::api::program_spl;
+use crate::integration_tests::api::program_uxd;
+
+pub async fn test_mint_with_identity_depository(
+    program_test_context: &mut ProgramTestContext,
+    program_keys: &program_uxd::accounts::ProgramKeys,
+    payer: &Keypair,
+    user: &Keypair,
+    user_collateral: &Pubkey,
+    user_redeemable: &Pubkey,
+    collateral_amount: u64,
+) -> Result<(), String> {
+    // Read state before
+    let redeemable_mint_before =
+        program_spl::accounts::read_token_mint(program_test_context, &program_keys.redeemable_mint)
+            .await?;
+
+    let controller_before =
+        program_uxd::accounts::read_controller(program_test_context, &program_keys.controller)
+            .await?;
+    let identity_depository_before = program_uxd::accounts::read_identity_depository(
+        program_test_context,
+        &program_keys.identity_depository_keys.depository,
+    )
+    .await?;
+
+    let redeemable_mint_supply_before = redeemable_mint_before.supply;
+    let redeemable_circulating_supply_before =
+        u64::try_from(controller_before.redeemable_circulating_supply).unwrap();
+    let redeemable_amount_under_management_before =
+        u64::try_from(identity_depository_before.redeemable_amount_under_management).unwrap();
+
+    let user_collateral_amount_before =
+        program_spl::accounts::read_token_account(program_test_context, user_collateral)
+            .await?
+            .amount;
+    let user_redeemable_amount_before =
+        program_spl::accounts::read_token_account(program_test_context, user_redeemable)
+            .await?
+            .amount;
+
+    // Execute
+    program_uxd::instructions::process_mint_with_identity_depository(
```

Comment:
Would remove the `program_uxd::instructions::process_mint_with_identity_depository` layer and put the transaction build/execution and the check of in/out values in this function

Reply:
Yeah thats a good point, maybe I should merge the two layers, but the problem is that it breaks general architechture:

program_credix/instructions // wrapper for credix instructions
program_mercurial/instructions // wrapper for mercurial instructions
program_uxd/instructions // this seems like a good place for wrapper for uxd instructions

Reply:
Yeah I think you're right, ill just put the asserts inside of program_uxd/instructions instead of creating the /case folder